### PR TITLE
fix: only use optionalReplacementSpan if client supports InsertReplace

### PR DESCRIPTION
**Pull request by rchl**
@predragnikolic

### CODE REVIEW

File: src/completion.ts [hunk 1]
```diff
@@ -63,7 +63,7 @@ export function asCompletionItem(entry: tsp.CompletionEntry, optionalReplacement
     }
 
     let insertText = entry.insertText;
-    const replacementSpan = entry.replacementSpan || optionalReplacementSpan;
+    const replacementSpan = entry.replacementSpan || (features.completionInsertReplaceSupport ? optionalReplacementSpan : undefined);
```
Comment (predragnikolic) [inline src/completion.ts:66]: I would suggest removing unnecessary code bellow:

```diff
- const replacementSpan = entry.replacementSpan || (features.completionInsertReplaceSupport ? optionalReplacementSpan : undefined);
+ const replacementSpan = entry.replacementSpan || optionalReplacementSpan;

...

- if (replacementRange) {
+ if (replacementRange && features.completionInsertReplaceSupport) {
    if (!insertText) {
        insertText = item.label;
    }
-    if (features.completionInsertReplaceSupport) {
      const insertRange = lsp.Range.create(replacementRange.start, position);
      item.textEdit = lsp.InsertReplaceEdit.create(insertText, insertRange, replacementRange);
-    } else {
-        item.textEdit = lsp.TextEdit.replace(replacementRange, insertText);
-    }
} else {
    item.insertText = insertText;
}
```

Comment (rchl) [reply src/completion.ts:66]: That doesn't sound right. The replacementRange can also come from the `entry.replaceSpan` in which case we want to create a replace TextEdit.

Comment (predragnikolic) [reply src/completion.ts:66]: True, you are right.

Comment (predragnikolic) [reply src/completion.ts:66]: Will try out this PR tomorrow.

**Comment by predragnikolic** [comment]
Just to document that this PR addresses this comment https://github.com/typescript-language-server/typescript-language-server/pull/583#issuecomment-1241795204

**Review by predragnikolic — APPROVED** [review]
I can verify that this PR brings back the same behavior as prior to InsertReplaceSupport for servers that do not have `insertReplaceSupport`.

**Pull request merged**